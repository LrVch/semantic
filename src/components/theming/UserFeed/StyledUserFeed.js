import {
  Card,
} from 'semantic-ui-react'
import styled from 'styled-components'

const StyledUserFeed = styled(Card)`
  &&& {
    box-shadow: none;

    .content:last-child {
      border-top-color: ${({ theme }) => theme.second};
    }
  }
`
export default StyledUserFeed