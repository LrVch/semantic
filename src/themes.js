export default {
  tomato: {
    name: 'tomato',
    main: 'tomato',
    second: '#ff6347ab',
    invert: 'white',
    dark: 'rgba(0,0,0,.87)'
  },
  palevioletred: {
    name: 'tomato',
    main: 'palevioletred',
    second: '#db709394',
    invert: 'white',
    dark: 'rgba(0,0,0,.87)'
  }
}