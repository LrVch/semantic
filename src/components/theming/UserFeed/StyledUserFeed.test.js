import 'jest-styled-components'

import { configure, shallow } from 'enzyme'

import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import StyledUserFeed from './StyledUserFeed'
import { mountWithTheme } from '../../../test'
import styled from 'styled-components'
import themes from '../../../themes'

configure({ adapter: new Adapter() })

describe('StyledUserFeed', () => {

  let wrapper
  let mounted

  beforeEach(() => {
    wrapper = shallow(<StyledUserFeed />)
    // mounted = mountWithTheme(<StyledUserFeed />, themes.tomato)
  })

  test('should create', () => {
    expect(wrapper).toBeTruthy()
  })

  test('applies defaulst and theme\'s styles ', () => {
    wrapper = shallow(<StyledUserFeed theme={{ second: 'fakeColor' }} />)

    // expect(wrapper).toMatchSnapshot()
    // expect(mounted).toHaveStyleRule('box-shadow', 'none', {
    //   modifier: '&&',
    // });
    // expect(mounted.find('.content:last-child')).toHaveStyleRule('background-color', '#fff', {
    //   modifier: '&&',
    // });
    // expect(wrapper).toHaveStyleRule('box-shadow', 'none', {
    //   modifier: '&&&'
    // })
    // expect(tree).toHaveStyleRule('margin-bottom', '15px', {
    //   modifier: '&&&'
    // })
  })
})


