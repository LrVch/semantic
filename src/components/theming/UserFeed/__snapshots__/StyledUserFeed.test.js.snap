// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledUserFeed applies defaulst and theme's styles  1`] = `ShallowWrapper {}`;
